Compose CLI sometimes returns 1 instead of 0 on success

**Description**
The Compose CLI sometimes returns 1 instead of 0 on a successful `docker compose up` update of an ECS stack.

Initial report was on the community Slack:
> I’m surprised by the error codes reported by docker compose -D up .  I would expect a successful deploy to return zero unless there is a failure.  My experience after several experiments is that when successfully deploying an update (such as simply changing the tag of a container) then the return code is 1.  Is this expected?  Is it documented somewhere?

> Hmm.. Sounds like a bug. Do you have a specific flow that you run and always get a 1 return code for?

> Yes.  I have consistently seen a one returned by docker compose -D -context mycontext up when the only change is the tag on one of the service images.  The deploy succeeds and the scope of changes reported by -D seems roughly appropriate, but the return code is always 1.

**Describe the results you expected:**
Return 0 on success and another value on error with a relevant error message.

Here is the output of an example where 1 is returned instead of 0:
```
ENVIRONMENT=production docker compose -D --context pipeline up --project-name pipeline
WARN[0000] services.deploy.restart_policy: unsupported attribute
WARN[0000] services.domainname: unsupported attribute
WARN[0000] services.expose: unsupported attribute
WARN[0000] services.hostname: unsupported attribute
WARN[0000] services.deploy.restart_policy: unsupported attribute
WARN[0000] services.domainname: unsupported attribute
WARN[0000] services.expose: unsupported attribute
WARN[0000] services.hostname: unsupported attribute
WARN[0000] services.deploy.restart_policy: unsupported attribute
WARN[0000] services.domainname: unsupported attribute
WARN[0000] services.expose: unsupported attribute
DEBU[0000] CheckRequirements on VPC : vpc-031993f2cefb56a87
DEBU[0000] Retrieve SubNets
DEBU[0000] Check if LoadBalancer exists: chat-lb
DEBU[0001] Create CloudFormation Changeset
[+] Running 2/0
 ⠿ pipeline                                            UpdateInProgress User Initiated                                                                                                                                      0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet0e5064a8380f41126  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet07f728e884d57606f  CreateComplete                                                                                                                                                       0.0s
 ⠋ PlineTaskRole                                       CreateComplete                                                                                                                                                       0.0s
 ⠋ Backend81Ingress                                    CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet019ce8543dc584fa5  CreateComplete                                                                                                                                                       0.0s
 ⠋ CloudMap                                            CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet07f728e884d57606f      CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet0ee63992acaf0bc86      CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet0e5064a8380f41126      CreateComplete                                                                                                                                                       0.0s
 ⠋ Pline82TargetGroup                                  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet0cd32172e0d13880e  CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet019ce8543dc584fa5      CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet0cd62e4e7a65d52e3  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet0ee63992acaf0bc86  CreateComplete                                                                                                                                                       0.0s
 ⠋ LogGroup                                            CreateComplete                                                                                                                                                       0.0s
 ⠋ HmacsecretSecret                                    DeleteComplete                                                                                                                                                       0.0s
 ⠋ Pline80TargetGroup                                  CreateComplete                                                                                                                                                       0.0s
 ⠋ Backend80Ingress                                    CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet019ce8543dc584fa5  CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionAccessPoint                                  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataAccessPoint                              CreateComplete                                                                                                                                                       0.0s
 ⠋ StoragedatomicpasswordSecret                        CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet0cd62e4e7a65d52e3      CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet0cd62e4e7a65d52e3  CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionNFSMountTargetOnSubnet0cd32172e0d13880e      CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsNFSMountTargetOnSubnet0cd32172e0d13880e  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet0ee63992acaf0bc86  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet07f728e884d57606f  CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomiclogsAccessPoint                              CreateComplete                                                                                                                                                       0.0s
 ⠋ Cluster                                             CreateComplete                                                                                                                                                       0.0s
 ⠋ Backend82Ingress                                    CreateComplete                                                                                                                                                       0.0s
 ⠋ StorageadminpasswordSecret                          CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicdataNFSMountTargetOnSubnet0e5064a8380f41126  CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionTaskExecutionRole                            CreateComplete                                                                                                                                                       0.0s
 ⠋ MailgunpasswordSecret                               DeleteComplete                                                                                                                                                       0.0s
 ⠋ Pline81TargetGroup                                  CreateComplete                                                                                                                                                       0.0s
 ⠋ Pline82Listener                                     CreateComplete                                                                                                                                                       0.0s
 ⠋ Pline80Listener                                     CreateComplete                                                                                                                                                       0.0s
 ⠋ Pline81Listener                                     CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicTaskExecutionRole                            CreateComplete                                                                                                                                                       0.0s
 ⠿ PlineTaskExecutionRole                              UpdateComplete                                                                                                                                                       0.0s
 ⠋ DatomicTaskRole                                     CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionTaskRole                                     CreateComplete                                                                                                                                                       0.0s
 ⠿ PlineTaskDefinition                                 DeleteComplete                                                                                                                                                       0.0s
 ⠋ BastionTaskDefinition                               CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicTaskDefinition                               CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionServiceDiscoveryEntry                        CreateComplete                                                                                                                                                       0.0s
 ⠋ PlineServiceDiscoveryEntry                          CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicServiceDiscoveryEntry                        CreateComplete                                                                                                                                                       0.0s
 ⠋ BastionService                                      CreateComplete                                                                                                                                                       0.0s
 ⠋ DatomicService                                      CreateComplete                                                                                                                                                       0.0s
 ⠿ PlineService                                        UpdateComplete                                                                                                                                                       0.0s
 ⠿ ConfigednSecret                                     UpdateComplete                                                                                                                                                       0.0s
Resource update cancelled
make: *** [.make.up] Error 1
```

--- a/ecs/up.go
+++ b/ecs/up.go
@@ -93,6 +93,16 @@ func (b *ecsAPIService) Up(ctx context.Context, project *types.Project, options 
 	if err != nil {
 		return err
 	}
+
+	var previousEvents []string
+	if update {
+		var err error
+		previousEvents, err = b.previousStackEvents(ctx, project.Name)
+		if err != nil {
+			return err
+		}
+	}
+
 	operation := stackCreate
 	if update {
 		operation = stackUpdate
@@ -121,6 +131,6 @@ func (b *ecsAPIService) Up(ctx context.Context, project *types.Project, options 
 		b.Down(ctx, project.Name, compose.DownOptions{}) // nolint:errcheck
 	}()
 
-	err = b.WaitStackCompletion(ctx, project.Name, operation)
+	err = b.WaitStackCompletion(ctx, project.Name, operation, previousEvents...)
 	return err
 }

